### PR TITLE
Fixing broken URLs

### PR DESCRIPTION
**Pull request by CRImier**
Was going through pages, found plenty of broken references. Ran a [broken URL checker](https://www.deadlinkchecker.com/website-dead-link-checker.asp) against the website, found a lot of broken URLs. Fixed some of them, esp. where syntax was concerned, in addition to that, fixed or removed some broken internal references. Also, replaced `site.com` references (real site) with `example.org`. Some URLs remain broken:

```
https://support.google.com/mail/forum/AAAAK7un8RU3J3r2JqFNTw/discussion/?hl=en&gpf=d/topic/gmail/3J3r2JqFNTw/discussion
https://www.javaworld.com/javaworld/javaqa/2003-05/01-qa-0509-jcrypt.html?page=2
http://www.php-security.org/downloads/rips.pdf
http://www.seclab.tuwien.ac.at/papers/pixy.pdf
http://w2spconf.com/2010/papers/p27.pdf
https://www.codemagi.com/blog/post/194
https://www.itu.int/rec/T-REC-X.690-200811-I/en
https://www.ietf.org/id/draft-ietf-websec-key-pinning-09.txt
https://github.com/andresriancho/w3af/blob/master/plugins/grep/csp.py
http://blog.php-security.org/archives/76-Holes-in-most-preg_match-filters.html
http://www.webapptest.org/ms-access-sql-injection-cheat-sheet-EN.html
http://labs.idefense.com/intelligence/vulnerabilities/display.php?id=77
http://www.ruxcon.org.au/files/2008/Attacking_Rich_Internet_Applications.pdf
http://yehg.net/lab/pr0js/files.php/inspath.zip
http://yehg.net/lab/pr0js/files.php/php_brute_force_detect.zip
http://www.comptechdoc.org/independent/web/cgi/ssimanual/ssiexamples.html
http://www.iss.net/security_center/advice/Exploits/TCP/session_hijacking/default.htm
http://www.derkeiler.com/pdf/Mailing-Lists/Securiteam/2002-12/0099.pdf
http://archives.neohapsis.com/archives/bugtraq/2002-05/0118.html
http://hacker-eliminator.com/trojansymptoms.html
http://www.microsoft.com/technet/security/bulletin/MS00-078.mspx
https://www.checkmarx.com/Demo/XSHM.aspx
https://blog.watchfire.com/wfblog/2008/06/javascript-code.html
http://shlang.com/netkill/netkill.html
https://cirt.net/code/nikto.shtml
https://addons.mozilla.org/en-US/firefox/addon/heartbleed-checker/
https://www.ecrimelabs.com/tools/webroot/WebRoot.txt
https://www.cs.rice.edu/~scrosby/hash/slides/USENIX-RegexpWIP.2.ppt
https://www.checkmarx.com/NewsDetails.aspx?id=23&cat=3
https://owasp.org/index.php/Dhiraj_Mishra
http://puzzlemall.googlecode.com/files/Session
https://owasp.org/index.php/Image:RequestRodeo-MartinJohns.pdf
http://windows.stanford.edu/docs/IISsecchecklist.htm
http://www.net-security.org/dl/articles/php-file-upload.pdf
http://www.windowsitpro.com/Files/18/27072/Webtable_01.pdf
https://www.imperva.com/404?aspxerrorpath=/application_defense_center/glossary/forceful_browsing.html
http://info.sen.ca.gov/pub/01-02/bill/sen/sb_1351-1400/sb_1386_bill_20020926_chaptered.html
https://blog.shapesecurity.com/heartbleed-bug-places-encrypted-user-data-and-webservers-at-risk
https://www.mitre.org/sites/default/files/publications/pr-18-2417-deliver-uncompromised-MITRE-study-8AUG2018.pdf
http://www.microsoft.com/technet/security/bulletin/ms04-028.mspx
http://www.digitaldwarf.be/products/mangle.c
http://projects.info-pull.com/mokb/
http://www.bonsai-sec.com/en/research/untidy-xml-fuzzer.php
https://support.snyk.io/snyk-cli/how-can-i-set-a-snyk-cli-project-as-open-source
http://www.rubcast.rub.de/index2.php?id=1009
http://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-53r4.pdf
http://aeditor.rubyforge.org/ruby_cplusplus/index.html
https://owasp-skf.gitbook.io/asvs-write-ups/filename-injection
http://tomcat.apache.org/tomcat-6.0-doc/config/context.html
https://blog.48bits.com/2010/09/28/iis6-asp-file-upload-for-fun-and-profit/
http://palisade.plynt.com/issues/2006Jun/injection-stored-procedures/
http://www.bindshell.net/tools/odysseus
http://www.ntobjectives.com/products/firewater/
http://home.intekom.com/rdawes/exodus.html
http://www.wastelands.gen.nz/odysseus/index.php
http://www.webcohort.com/web_application_security/research/tools.html
http://www.rsasecurity.com/standards/ssl/basics.html
http://palisade.plynt.com/issues/2005Aug/page-tokens/
http://www.microsoft.com/mspress/books/toc/5612.asp
http://www.seczone.cn/2018/06/27/codesec源代码安全检测平台/
```

If anyone wants to go through these, `grep --color=always -nr -Ff broken_urls_left.txt|grep --color=always -v "broken_"|sort` will show where those URLs are specifically. Could probably also find  a lot of broken internal references by looking for "wikilink".

Please review before merge.

### CODE REVIEW

File: pages/Fuzzing.md [hunk 1]
```diff
@@ -184,7 +184,7 @@ appear these days; some examples:
 
 <!-- end list -->
 
-  - One may use tools like [Hachoir](http://hachoir.org/) as a generic
+  - One may use tools like Hachoir as a generic
```
Comment (kingthorin) [inline pages/Fuzzing.md:187]:
```suggestion
  - One may use tools like [Hachoir](https://hachoir.readthedocs.io/) as a generic
```

File: pages/attacks/Blind_SQL_Injection.md [hunk 2]
```diff
@@ -196,12 +196,12 @@ vulnerabilities.
 - Kevin Spett from SPI Dynamics:
   - http://www.net-security.org/dl/articles/Blind_SQLInjection.pdf
   - http://www.imperva.com/resources/whitepapers.asp?t=ADC
-  - [Advanced SQL Injection](https://www.owasp.org/images/7/74/Advanced_SQL_Injection.ppt)
+  - [Advanced SQL Injection](https://www.linuxsecrets.com/owasp-wiki/images/7/74/Advanced_SQL_Injection.ppt)
```
Comment (kingthorin) [inline pages/attacks/Blind_SQL_Injection.md:199]:
```suggestion
  - [Advanced SQL Injection](https://wiki.owasp.org/images/7/74/Advanced_SQL_Injection.ppt)
```

File: pages/attacks/Cross-User_Defacement.md [hunk 3]
```diff
@@ -89,7 +89,7 @@ This way it was possible to replace the web page, which was served to
 the specified user.
 
 More information can be found in one of the presentations under
-<http://www.owasp.org/images/1/1a/OWASPAppSecEU2006_HTTPMessageSplittingSmugglingEtc.ppt>
+<https://www.linuxsecrets.com/owasp-wiki/images/1/1a/OWASPAppSecEU2006_HTTPMessageSplittingSmugglingEtc.ppt>
```
Comment (kingthorin) [inline pages/attacks/Cross-User_Defacement.md:92]:
```suggestion
<http://wiki.owasp.org/images/1/1a/OWASPAppSecEU2006_HTTPMessageSplittingSmugglingEtc.ppt>
```

File: pages/attacks/DOM_Based_XSS.md [hunk 4]
```diff
@@ -131,7 +131,7 @@ Applications from Universal PDF XSS: A discussion of how weird the web
 application security world has become" at the [2007 OWASP Europe AppSec
 Conference](OWASP_AppSec_Europe_2007_-_Italy "wikilink") in Milan. The
 presentation (\[5\]) can be downloaded
-[here](http://www.owasp.org/images/c/c2/OWASPAppSec2007Milan_ProtectingWebAppsfromUniversalPDFXSS.ppt).
+[here](https://www.linuxsecrets.com/owasp-wiki/images/c/c2/OWASPAppSec2007Milan_ProtectingWebAppsfromUniversalPDFXSS.ppt).
```
Comment (kingthorin) [inline pages/attacks/DOM_Based_XSS.md:134]:
```suggestion
[here](https://wiki.owasp.org/images/c/c2/OWASPAppSec2007Milan_ProtectingWebAppsfromUniversalPDFXSS.ppt).
```

File: pages/attacks/DOM_Based_XSS.md [hunk 5]
```diff
@@ -202,7 +202,7 @@ Giorgio Fedon, December 2006
 \[5\] "Protecting Web Applications from Universal PDF XSS" (2007 OWASP
 Europe AppSec presentation) Ivan Ristic, May 2007
 
-<http://www.owasp.org/images/c/c2/OWASPAppSec2007Milan_ProtectingWebAppsfromUniversalPDFXSS.ppt>
+<https://www.linuxsecrets.com/owasp-wiki/images/c/c2/OWASPAppSec2007Milan_ProtectingWebAppsfromUniversalPDFXSS.ppt>
```
Comment (kingthorin) [inline pages/attacks/DOM_Based_XSS.md:205]:
```suggestion
<https://wiki.owasp.org/images/c/c2/OWASPAppSec2007Milan_ProtectingWebAppsfromUniversalPDFXSS.ppt>
```

File: pages/attacks/SQL_Injection_Bypassing_WAF.md [hunk 6]
```diff
@@ -120,9 +120,8 @@ can be used (e.g., \#\#\#\#\#, %00).*
 ` /?id=1;select+1&id=2,3+from+users+where+id=1--`
 
 *Successful conduction of an HPP attack bypassing WAF depends on the
-environment of the application being attacked.*
-*\[<http://www>..org/images/b/ba/AppsecEU09_CarettoniDiPaola_v0.8.pdf
-EU09 Luca Carettoni, Stefano diPaola\].*
+environment of the application being attacked.
+*[<http://www.owasp.org/images/b/ba/AppsecEU09_CarettoniDiPaola_v0.8.pdf](EU09 Luca Carettoni, Stefano diPaola)*
```
Comment (kingthorin) [inline pages/attacks/SQL_Injection_Bypassing_WAF.md:124]:
```suggestion
*[http://wiki.owasp.org/images/b/ba/AppsecEU09_CarettoniDiPaola_v0.8.pdf](EU09 Luca Carettoni, Stefano diPaola)*
```

File: pages/initiatives/code_sprint/wcs2014.md [hunk 7]
```diff
@@ -22,7 +22,7 @@ You make excellent contacts and you participate in an international team
 
 Students who successfully(*) participate in the project will get:
 
-* An OWASP annual individual membership. More info [here](http://owasp.com/index.php/Individual_Member)
+* An OWASP annual individual membership. More info [here](http://owasp.org/index.php/Individual_Member)
```
Comment (kingthorin) [inline pages/initiatives/code_sprint/wcs2014.md:25]:
```suggestion
* An OWASP annual individual membership. More info [here](https://owasp.org/membership/)
```

File: pages/vulnerabilities/Information_exposure_through_query_strings_in_url.md [hunk 8]
```diff
@@ -51,7 +51,7 @@ The following figure displays how an internal attacker can potentially exploit t
 
 ## References
 
-- [Testing for Exposed Session Variables (OTG-SESS-004)](https://www.owasp.org/index.php/Testing_for_Exposed_Session_Variables_\(OTG-SESS-004\))
+- [Testing for Exposed Session Variables (OTG-SESS-004)](https://www.owasp.org/index.php/Testing_for_Exposed_Session_Variables_%28OTG-SESS-004%29)
```
Comment (kingthorin) [inline pages/vulnerabilities/Information_exposure_through_query_strings_in_url.md:54]:
```suggestion
```

That's out dated and no longer relevant.

File: pages/vulnerabilities/Unrestricted_File_Upload.md [hunk 9]
```diff
@@ -487,9 +487,7 @@ And some special recommendations for the developers and webmasters:
   - Potentially Dangerous File Types
     [7](http://www.windowsitpro.com/Files/18/27072/Webtable_01.pdf)
   - Image Upload XSS
-    [8](http://ha.ckers.org/blog/20070603/image-upload-xss/)
```
Comment (kingthorin) [inline pages/vulnerabilities/Unrestricted_File_Upload.md:490]: The line before should be removed as well.

File: pages/vulnerabilities/Unrestricted_File_Upload.md [hunk 10]
```diff
@@ -487,9 +487,7 @@ And some special recommendations for the developers and webmasters:
...
   - Code Execution Through Filenames in Uploads
-    [9](http://ha.ckers.org/blog/20070620/code-execution-through-filenames-in-uploads/)
```
Comment (kingthorin) [inline pages/vulnerabilities/Unrestricted_File_Upload.md:492]: The line before should be removed as well.

**Review by kingthorin — CHANGES_REQUESTED** [review]

**Comment by CRImier** [comment]
All changes made. Would you like me to squash the changes and push them as one commit? Will force push so that there's no need to re-create the PR.

**Comment by kingthorin** [comment]
That'd be great, thanks!

**Comment by CRImier** [comment]
Done.

**Review by kingthorin — APPROVED** [review]
Looks good, thanks ⭐

**Pull request merged**